spec/models/manageiq/providers/openstack/cloud_manager/provision_workflow_spec.rb

```diff
@@ -321,6 +321,28 @@
     end
   end
 
+  describe "prepare_volumes_fields" do
+    let(:workflow) do
+      stub_dialog
+      allow_any_instance_of(described_class).to receive(:update_field_visibility)
+      described_class.new({:src_vm_id => template.id}, admin.userid)
+    end
+    it "converts numbered volume form fields into an array" do
+      volumes = workflow.prepare_volumes_fields(
+        :volume_name_1 => "v1n", :volume_size_1 => "v1s", :volume_delete_on_terminate_1 => true,
+        :volume_name_2 => "v2n", :volume_size_2 => "v2s", :volume_delete_on_terminate_2 => false,
+        :other_irrelevant_key => 1
+      )
+      expect(volumes.length).to eq(2)
+      expect(volumes[0]).to eq(:volume_name => "v1n", :volume_size => "v1s", :volume_delete_on_terminate => true)
+      expect(volumes[1]).to eq(:volume_name => "v2n", :volume_size => "v2s", :volume_delete_on_terminate => false)
```

Comment:
@mansam Backend code expects volume keys :name and :size (not :volume_name and :volume_size in this part of code).